statesync/reactor_test.go

```diff
@@ -1,21 +1,85 @@
 package statesync
 
 import (
+	"context"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
 
 	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tendermint/tendermint/libs/log"
 	"github.com/tendermint/tendermint/p2p"
-	p2pmocks "github.com/tendermint/tendermint/p2p/mocks"
 	ssproto "github.com/tendermint/tendermint/proto/tendermint/statesync"
 	proxymocks "github.com/tendermint/tendermint/proxy/mocks"
 )
 
-func TestReactor_Receive_ChunkRequest(t *testing.T) {
+type ReactorTestSuite struct {
```

Comment:
ACK. Since testify won't allow use to execute setup/teardown on a subtest-basis, I'll revert the usage of suite here in favor of something like you've proposed.